test/reifyhealth/specmonstah/core_test.cljc

```diff
@@ -540,14 +540,25 @@
                    (lat/add-attr :t0 :query-term [:t0 {:refs {:todo-list-id :tl0}}])
                    (lat/add-attr :t0 :ent-type :todo)
                    (lat/add-attr :t0 :tl0 :relation-attrs #{:todo-list-id})
-                   
+
                    (lat/add-attr :todo-list :type :ent-type)
                    (lat/add-attr :tl0 :type :ent)
                    (lat/add-attr :tl0 :index 0)
                    (lat/add-attr :tl0 :query-term [:tl0 {:refs {:first-todo-id :t0}}])
                    (lat/add-attr :tl0 :ent-type :todo-list)
                    (lat/add-attr :tl0 :t0 :relation-attrs #{:first-todo-id})))))
 
+(deftest test-add-ents-handles-A->self-cycles
+  (testing "Handle cycles where an entity references itself"
+    (is-graph= (:data (sm/add-ents {:schema td/cycle-schema} {:user [[:u0 {:refs {:updated-by-id :u0}}]]}))
+               (-> (lg/digraph [:user :u0] [:u0 :u0])
+                   (lat/add-attr :user :type :ent-type)
+                   (lat/add-attr :u0 :type :ent)
+                   (lat/add-attr :u0 :index 0)
+                   (lat/add-attr :u0 :query-term [:u0 {:refs {:updated-by-id :u0}}])
+                   (lat/add-attr :u0 :ent-type :user)
+                   (lat/add-attr :u0 :u0 :relation-attrs #{:updated-by-id})))))
+
```

Comment:
This test will pass even without the change in core, this is just to make sure the functionality doesn't break.

Reply:
that's the test below, `test-visit-self-reference`. It's the visiting part that was actually broken.

This is just adding coverage for some functionality that's required for self references to work even though this part didn't require a change

Reply:
oh great, sorry I misunderstood!